src/script/api/script_date_economy.cpp

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file script_date_economy.cpp Implementation of ScriptDateEconomy. */
+
+#include "../../stdafx.h"
+#include "script_date_economy.hpp"
+#include "../../timer/timer_game_economy.h"
+
+#include <time.h>
+
+#include "../../safeguards.h"
+
+/* static */ bool ScriptDateEconomy::IsValidDate(Date date)
+{
+	return date >= 0;
+}
+
+/* static */ ScriptDateEconomy::Date ScriptDateEconomy::GetCurrentDate()
+{
+	return (ScriptDateEconomy::Date)TimerGameCalendar::date.base();
```

Comment:
I know you plan to do it differently, but looks like it's using the wrong timer, and it's the same in all other functions (except for parameter types).

Reply:
Oops, faulty (user of) find-and-replace. But yes, I'm going to do this the other way.